cmd/flow-dps-client/main.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021 Alvalor S.A.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/rs/zerolog"
+	"github.com/spf13/pflag"
+	"google.golang.org/grpc"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/encoding/json"
+
+	"github.com/optakt/flow-dps/api/dps"
+	"github.com/optakt/flow-dps/rosetta/invoker"
+	"github.com/optakt/flow-dps/rosetta/lookup"
+	"github.com/optakt/flow-dps/rosetta/read"
+)
+
+func main() {
+
+	// Signal catching for clean shutdown.
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt)
+
+	// Command line parameter initialization.
+	var (
+		flagAPI    string
+		flagHeight uint64
+		flagLog    string
+		flagParams string
+		flagScript string
+	)
+
+	pflag.StringVarP(&flagAPI, "api", "a", "127.0.0.1:5005", "host for GRPC API server")
+	pflag.Uint64VarP(&flagHeight, "height", "h", 0, "block height to execute the script at")
+	pflag.StringVarP(&flagLog, "log", "l", "info", "log output level")
+	pflag.StringVarP(&flagParams, "params", "p", "", "JSON encoded Cadence parameters for the script")
```

Comment:
I think it's optional.

Reply:
What do you mean? Having a hyphen between compound words used as adjectives?

Here `JSON-encoded` is an adjective for `Cadence parameters`, and since `JSON` and `encoded` are two distinct words, there should be a hyphen between them.

[Source 1](https://www.grammarly.com/blog/hyphen/)

There are indeed exceptions but it is not the case here I'm pretty sure. [Source 2](https://en.wikipedia.org/wiki/Compound_modifier#Exceptions)

Even the [javascript official docs use a hyphen for this exact case](https://javascript.info/json).

Reply:
I think it depends on whether you use American or British English.